guides/covariates_tfms.md

```diff
@@ -0,0 +1,289 @@
+
+So during training, the torch models will go through the training data in sequences (see Figure 3). Using information from the **input chunk** and **output chunk**, the model predicts the future target on the output chunk. The training loss is evaluated between the predicted future target and the actual target value on the output chunk. The model trains itself by minimizing the loss over all sequences.
+
+![figure5](./images/covariates/seq_covs_1.png)
+
+**Figure 3: Prediction and loss evaluation in a single sequence**
+
+After having completed computations on the first sequence, the model moves to the next sequence, which, in the
+case of sequential models, starts one time step later than the previous sequence. This process is repeated until all 365 days were covered (see Figure 4).
+
+![figure6](./images/covariates/sequential_training.png)
+
+**Figure 4: Move to next sequence and repeat**
+
+### 2.1.1. Training with a validation dataset
+
+You can also train your models with a validation dataset:
+
+```
+# create train and validation sets
+ice_cream_sales_train, ice_cream_sales_val = ice_cream_sales.split_after(training_cutoff)
+
+# train with validation set
+model.fit(series=ice_cream_sales_train,
+          past_covariates=temperature, 
+          future_covariates=weekday,
+          val_series=ice_cream_sales_val,
+          val_past_covariates=temperature,
+          val_future_covariates=weekday)
+```
+
+If you split your data, you have to define a `training_cutoff` (a date or fraction at which to split the dataset) so that both the train and validation datasets satisfy the minimum length requirements
+from [section 1.3.](#13-required-target-time-spans-for-training-validation-and-prediction)
+
+Instead of splitting by time, you can also use another subset of time series as validation set.
+
+The model trains itself the same way as before but additionally evaluates the loss on the validation dataset. If you want to keep track of the best performing model on the validation set, you have to enable checkpoint saving as shown next.
+
+### 2.1.2. Saving and Loading Model States
+Per default, the models don't automatically save any checkpoints. If you want to keep track of the best performing model on the validation set and the latest 5 epochs, you have to enable checkpoint saving at model creation:
+
+```
+model = SomeTorchForecastingModel(..., model_name='MyModel', save_checkpoints=True)
+
+# checkpoints are saved automatically
+model.fit(...)
+
+# load the model state that performed best on validation set 
+best_model = model.load_from_checkpoint(model_name='MyModel', best=True)
+```
+
+You can also save or load manually:
+```
+model.save_model(model_path)
+loaded_model = model.load_model(model_path)
+```
+
```

Comment:
```suggestion
/!\ Warning /!\ At this stage of Darts development, we are not (yet) ensuring backward compatibility, so it might not always be possible to load a model saved by an older version of the library.
```